### ISSUE TITLE
SA1018 incorrectly reported at beginning of line

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#258]
The diagnostic for SA1018 does not correctly determine whether a token is the first in the line or not. The following code produces an incorrect SA1018 warning for the `?` token. Note that this is only reproduced when the code is not indented.

```csharp
int
? data;
```
